packages/http-client/src/proxy.ts

```diff
@@ -51,7 +51,7 @@ export function checkBypass(reqUrl: URL): boolean {
     .split(',')
     .map(x => x.trim().toUpperCase())
     .filter(x => x)) {
-    if (upperReqHosts.some(x => x === upperNoProxyItem)) {
+    if (upperReqHosts.some(x => x.includes(upperNoProxyItem))) {
```

Comment:
I think it would be safer to use `endsWith` instead of `includes` to avoid bad actors from bypassing the proxy. For example if we set `no_proxy=mycompany.com`  then a domain like `mycompany.com.evil.org` should not bypass the proxy.

```suggestion
    if (upperReqHosts.some(x => x.endsWith(upperNoProxyItem))) {
```
The above would still let domains like `evilmycompany.com` bypass the proxy. Perhaps a better formulation would be
 
```suggestion
    if (upperReqHosts.some(x => x === upperNoProxyItem || x.endsWith('.' + upperNoProxyItem))) {
```

In addition we might want to strip off a leading `.`  to ensure that `no_proxy=example.com` and `no_proxy=.example.com` are treated the same.

Reply:
+1 for @aibaars suggestions too, ensWith + '.' sounds good
NIT: I'd consider a small refactoring of the block from line 54, it's becoming a bit dense with filters

@felixlut Could you add a test-case that covers the change in https://github.com/actions/toolkit/blob/main/packages/http-client/__tests__/proxy.test.ts ?

PS the PR validation currently fails due to some older packages, I'll take a look in a different PR

Reply:
I like your suggestions, and sure I can add a test-case. I'll try to find time over the weekend!

Reply:
I've implemented the suggested changes (in addition to some other suggestions from the [article](https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/#standardizing-no_proxy) I posted above), as well as test-cases for each of them. Namely the following was added:
* `*` matching all hosts
* Strip leading dots (`.domain.com` --> `domain.com`)